src/data/nat/fib/zeckendorf.lean

```diff
@@ -0,0 +1,229 @@
+/-
+Copyright (c) 2022 Jakub Kądziołka. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakub Kądziołka
+-/
+import data.nat.fib.basic
+import algebra.big_operators.order
+import tactic.omega.main
+import tactic.linarith
+
+/-!
+# Zeckendorf's Theorem
+
+`nat.zeckendorf`: Each natural number can be written in exactly one way as a sum of distinct,
+non-consecutive Fibonacci numbers.
+-/
+
+open_locale big_operators
+
+namespace nat
+
+/-- The set `S` contains the indices which have been chosen.
+    For uniqueness, we need to exclude `fib 0` and `fib 1`,
+    since `fib 0 = 0` and `fib 1 = fib 2 = 1`. -/
+def is_zeckendorf_rep (S : finset ℕ) : Prop :=
+  ∀ (n ∈ S), n ≥ 2 ∧ n + 1 ∉ S
+
+/-- Every natural number lies between two Fibonacci numbers. -/
+lemma fib_between (n : ℕ): ∃ k : ℕ, fib k ≤ n ∧ n < fib k.succ :=
```

Comment:
This really is a result about strictly monotone functions. `fib` is not strictly montone, but `λ n, fib (n + 1)` is, so you can use that.